pandas/core/frame.py

```diff
@@ -7803,11 +7804,11 @@ def _reduce(
         self, op, name, axis=0, skipna=True, numeric_only=None, filter_type=None, **kwds
     ):
 
-        dtype_is_dt = self.dtypes.apply(lambda x: x.kind == "M")
+        dtype_is_dt = self.dtypes.apply(lambda x: x.kind == "M" or is_period_dtype(x))
```

Comment:
can you apply needs_i8_conversion instead here? (like you do below), this is a non-standard usage

Reply:
trouble is we don't want to include td64

Reply:
ok how about using is_datetime64_any_dtype then; this is a non-obvious pattern